explainer-datamodel.md

```diff
@@ -1,236 +1,329 @@
+    overlapping formatting.
+* Extensibility for internationalization hints and reserved future scenarios.
 
-### Primary Objects
+## Formatting text (`FormattedText.format`)
 
-The data model for formatted text is quite simple and involves only two objects: `FormattedText`
-is a container that orders a list of `FormattedTextRun` objects that contain the actual text and
-formatting information. (The `FormattedText` container can also have formatting applied that may
-inherit down to the `FormattedTextRun`s.)
+The data model comprises the parameters to the operation `format` which takes the input data model
+and produces output [text metrics](explainer-metrics.md) (`format` is a static method of the
+`FormattedText` constructor). The input consists of:
 
-To maximize authoring convenience, we are proposing using the 
-[ObserverableArray pattern](https://heycam.github.io/webidl/#idl-observable-array) to allow authors
-to manage the contained array exactly like a JavaScript Array.
+1. text or an array of text
+2. metadata/ style for all the text
+3. constraints for the formatting process
 
-<img src="explainerresources/formatted-text-om.png" alt="FormattedText object holds an array of FormattedTextRun objects through a property called 'textruns'." align="center"/>
+Each of these is explained in more detail below.
 
-## Creating FormattedText and FormattedTextRun objects
+### Expressing text
 
-`FormattedText` and `FormattedTextRun` objects are created using the 'new' pattern:
+Text can be expressed as JavaScript strings, or as a JavaScript object with property named
+`text` that has a JavaScript string as its value.
 
 ```js
-let text = new FormattedText();
-let textrun = new FormattedTextRun();
+// The following representations of text are equivalent:
+FormattedText.format( "hello" );
+FormattedText.format( { text: "hello" } );
 ```
 
-Attach `FormattedTextRun` objects to their `FormattedText` object's `textruns` property (an Array):
+Text can also be expressed in separate runs (which are concatenated together when formatted)
+when wrapped in an Array. The following are equivalent:
 
 ```js
-text.textruns.push( textrun );
+// The following text runs are equivalent:
+FormattedText.format( [ "hello", " world!" ] );
+FormattedText.format( [ { text: "hello" }, { text: " world!" } ] );
 ```
 
-Add text to the `FormattedTextRun` object:
+### Adding some style
+
+Style for *all the text* (general styles) are provided via the second parameter to `format`.
+Style input uses the same syntax as Element inline styles 
+(i.e., `<span style="style text syntax">`) and is ultimately parsed with the same CSS parser.
+The following formatted text outputs will all be blue:
 
 ```js
-textrun.text = "hello";
+FormattedText.format( "hello world!", "color:blue" );
+FormattedText.format( [ "hello", " world!" ], "color: blue" );
+FormattedText.format( [ { text: "hello" }, { text: " world!" } ], " color :blue" );
```

Comment:
...yeah, actually that was intentional--the extra whitespace placement, etc. (but probably too subtle). :) I'll make it consistent since it's mostly just distracting.